packages/next/src/build/analysis/get-page-static-info.ts

```diff
@@ -142,8 +146,27 @@ function checkExports(
       let generateSitemaps: boolean = false
       let generateStaticParams = false
       let extraProperties = new Set<string>()
+      let directives = new Set<string>()
+      let hasLeadingNonDirectiveNode = false
 
       for (const node of swcAST.body) {
+        // There should be no non-string literals nodes before directives
+        if (
+          node.type === 'ExpressionStatement' &&
+          node.expression.type === 'StringLiteral'
+        ) {
+          if (!hasLeadingNonDirectiveNode) {
+            const directive = node.expression.value
+            if (CLIENT_DIRECTIVE === directive) {
+              directives.add('client')
+            }
+            if (SERVER_ACTION_DIRECTIVE === directive) {
+              directives.add('server')
+            }
```

Comment:
I just thought of something: 

What happens when the file has `use client` followed by `use server`?

Reply:
It will also error with compiling, it's not allowed to have both, we have test cases covered in `next-swc`